Spring-2019/trees_graphs/SOLUTIONS_2019Spring.md

```diff
@@ -76,6 +76,40 @@ An example of a binary search tree that contains characters is:
 
 ### What is a graph?
 
+A graph is also a data structure that organizes and shows relationships 
+between individual data nodes. However, a graph is not organized in a 
+heirarchal way the way a tree is. Nodes in a graph are not required 
+to be connected, and connections (or _edges_) between nodes can be 
+created and destroyed dynamically.
+
+Graphs are meaningful in the context of computer science because they 
+are an excellent model for social interactions. Most social media interactions 
+can be represented using graphs.
+
+#### Undirected Graphs
+
+![facebook as an undirected graph](./background/facebook-undirected.png)
+
+#### Directed Graphs
+
+![twitter as a directed graph](./background/twitter-directed.png)
+
+#### Weighted Graphs
```

Comment:
Are there problems that will be presenting on weighted graphs? This section might not be needed if we aren't.

Reply:
I will put some more thought into it. I have a (perhaps overly ambitious) plan to do Kevin Bacon using a weighted graph. I'll get back to you on this one

Reply:
So, I'm giving the problem as a weighted graph, so I'm inclined to stick with this section in the background section :grin: